server/services/zwave/lib/comClass/defaultCommandClass.js

```diff
@@ -0,0 +1,83 @@
+const { transformValue, normalizeValue } = require('../utils/valueBinders');
+
+const DefaultCommandClass = function DefaultCommandClass() {};
+
+DefaultCommandClass.prototype = {
```

Comment:
Why didn't you use functions? I don't really approve the use of this design pattern here. 

Just write the function and exports them, the simpler the better!

Reply:
Because they are not just "functions" it's a real object with real behaviors. It's not just a set of tools or utils. It represents an object with a set of functionnalities. And it's actually how Javascript is working for this. Finality here is to have the possibility to get one fully representative object per zWave CommandClass to handle specific cases on specific CommandClasses.

Look at how the factory is working, it will make more sense than functions.

Reply:
I read the code, I don't see the added value in using POO here. It's a singleton class with only static functions, it could just be functions.

Reply:
They are not static methods. 

Writing this comment... I just think about one thing: I maybe mixed two things in your comment (or mixed the two comments, my bad): code splitting and POO (the other comment).

This code commented here is exactly the same thing as here: https://github.com/GladysAssistant/Gladys/blob/master/server/services/zwave/lib/index.js#L64 (and all over the project). So, if your point is to explode each function in a dedicated file to finaly put it in the prototype: this code is doing exactly the same thing... but in a single file. It's far most easier to navigate in your class with a single file (knowing which properties are available) than in one with multiple files (for example, when I'm in the file [zwave.setValue](https://github.com/GladysAssistant/Gladys/blob/4286e003c1a46e49463149dd95dc338404f4ff15/server/services/zwave/lib/commands/zwave.setValue.js#L15) `this.zWave` doesn't mean anything. It's totally out of it context.


About the POO, the other comment, look at the usage of the getMinMax value. Without it, you would have a function growing over and over with command classes. The function would consist of a big `switch()`.

Look at the code when a "value changed" event is received. Without POO, you will also need a function based on switch and functions and so on... It's not relevant enough now because we have only the SwitchMultiLevelCommandClass implemented. With time, you will have more and more.



You are the owner of the project, you will have the final word. 
So, about splitting the code in multiple files to put all of them in the prototype in a `index.js` file... Whatever... if you want... but result is exactly the same: there is no design pattern difference here... This is far most usual Javascript than using export everywhere to associate them to the prototype (to be honest, when I started to deep dive in Gladys and see this, I've been surprised... but OK, result is exactly the same). But if you absolutely want to split, I'll do.

For inheritance, I'm convinced this is a long term advantage and avoid errors in coming CommandClasses implementations. There is no magick here, just usual Javascript.

Reply:
> They are not static methods.

They are not using "this". They are just using function parameters to return a result, it's just utils functions. 

>  result is exactly the same

Yes, but it'll be readable. Right now, yes it works but it's hard to understand and it'll lead to weird bugs in the future. 

Inheritance is debatable, to me it'll just make the maintenance more complex in this case.

From what I see, your function are just "utils" functions, it could be a simple file with functions and you export the functions. 

**About the singleton problem**

I don't agree, it's not the same as others design patterns used. You're exporting the object, not the class.

<img width="305" alt="Screenshot 2020-09-03 at 11 51 28" src="https://user-images.githubusercontent.com/7365207/92100288-e7c09480-eddb-11ea-8c0a-4fd693c1c258.png">

In short: you're using the filesystem (require) as a way to export a singleton.

In Node.js, when you do a "require" on a file, the result is cached. So if N files are requiring your file, just one instance of your class will be created. 

This is not right, because it means the object is now a global, and tests are sharing a global variable. 

In the example you take, I'm exporting the class, not an object of the class.

<img width="244" alt="Screenshot 2020-09-03 at 11 54 54" src="https://user-images.githubusercontent.com/7365207/92100811-8b11a980-eddc-11ea-8680-d528b49e4f7c.png">


> Let me know

Just do a simple file with functions and call the functions: simpler, easier to read & maintain.

Reply:
Even if I don't share all your points, you're the boss.

As soon as I have time, I'll update the PR.

Reply:
We can always debate about that in a community call, it'll be more easy to talk about that vocally ;) 

> As soon as I have time, I'll update the PR.

Thank you very much for your time and contribution to Gladys 🙏 Don't take personally my feedbacks, computer sciences is a topic full of controversy and everyone has different theory. It's great to have such discussion in PR 🙂